src/__tests__/ObjectStateMutations-test.js

```diff
@@ -313,16 +313,6 @@ describe('ObjectStateMutations', () => {
     expect(serverData).toEqual({ items: [{ count: 15 }, { count: 4 }] });
   });
 
-  it('can commit nested json array changes from the server to empty serverData', () => {
-    const serverData = {};
-    const objectCache = {};
-    ObjectStateMutations.commitServerChanges(serverData, objectCache, {
-      items: { '0': { count: 20 }, '1': { count: 5 } },
-    });
-    expect(serverData).toEqual({ items: [{ count: 20 }, { count: 5 }] });
-    expect(objectCache).toEqual({ items: '[{"count":20},{"count":5}]' });
-  });
-
```

Comment:
@mtrezza I would keep this to show that the JSON array doesn't transform

```
it('can commit nested json array changes from the server to empty serverData', () => {
    const serverData = {};
    const objectCache = {};
    ObjectStateMutations.commitServerChanges(serverData, objectCache, {
      items: { '0': { count: 20 }, '1': { count: 5 } },
    });
    // Should not transform
    expect(serverData).toEqual({ items: { '0': { count: 20 }, '1': { count: 5 } } });
    expect(objectCache).toEqual({ items: '{"0":{"count":20},"1":{"count":5}}' });
  });
```

Reply:
This test fails, that's why I've removed it

Reply:
Look again I updated it

Reply:
Added the new test